### ISSUE TITLE
[Feature] Customizable Date range for scheduled transactions in bank account

### ISSUE THREAD

**Issue by msmith8fti** [actualbudget/actual#1890]
### Verified feature request does not already exist?

- [X] I have searched and found no existing issue

### 💻

- [X] Would you like to implement this feature?

### Pitch: what problem are you trying to solve?

In an account, it does currently show a number of scheduled transactions - I believe up to one week. It would be beneficial to be able to customize the number of upcoming transactions to see in the account register. 

### Describe your ideal solution to this problem

- A custom setting to configure the date range
- The ability to directly add a transaction from the schedule to the account; with the schedule showing the transaction as pending

**Comment by rickdoesdev** [comment]
This is something I've been keen to have as well and I did start digging into the logic for this a couple months ago but had to pause it.

I am probably going to try to pick this feature up over the holiday period when I have free time.

Once I have a better understanding of the schedule and transaction table and life cycle I'll post a proposed set of mockups for new experience and get some feedback.

One thing that needs to be addressed is that currently schedules that repeat within that preview period don't show the repeats. This would be critical for my own needs for extending the scheduled duration. I will add a separate issue for that and link these two together as that one probably needs to be fixed before this one.

From memory, (correct me if I'm wrong anyone who knows better) when I looked into this before the scheduled transactions are actually stored as records in the DB themselves, and not calculated in real time (sensible for performance) but this means if the schedule preview window is configurable we would have to generate or delete new schedule entries every time the configured value changes. This would probably preclude things like being able to change the duration from the accounts view, as nice as that would be.
